sharktank/tests/ops/sharded_test.py

```diff
@@ -667,10 +667,10 @@ def testReplicatedLhsShardedParallelDimRhs(self):
         b = torch.rand(3, 6, dtype=torch.float32)
         shard_count = 3
         unsharded_result = torch.matmul(a, b)
-        expected_result = ops.reshard_split(unsharded_result, dim=2, count=shard_count)
+        expected_result = ops.reshard_split(unsharded_result, dim=2, count=shard_count)  # TODO: How to know this should also not be pinned
         b_sharded = ops.reshard_split(b, dim=1, count=shard_count)
         a_sharded = ops.replicate(a, count=shard_count)
-        actual_result = ops.matmul(a_sharded, b_sharded)
+        actual_result = ops.matmul(a_sharded, b_sharded)  # GOOD: Should NOT be pinned
         assert ops.equal(expected_result, actual_result)
```

Comment:
Adding `.pinned` and `.devices` to `is_deep_equals`, which `ops.equals` calls, causes this test to fail.

`actual_result.pinned == False` which is correct. But `expected_result.pinned` would end up being `True` if any heuristic is used to convert the default `None` value into a bool: it's a concrete `torch.Tensor` and would be indistinguishable from one loaded from a file, i.e. a weight.

How to handle this? Should `is_deep_equal` be looking at `.devices` and `.pinned`? I feel like it should given its docstring and current behaviour.

Reply:
Blah, I can see both cases. We may want to update to include an option to include comparisons of pinning. `ops.equals` is such a special case that I can see us wanting to compare both numerics and placement information. For our tests though we mostly just want to compare numerics and pinning comparison is mostly metadata and not value. If this is the only case we can always just manually compare.